docs/en/guides/testing-async-components.md

```diff
@@ -0,0 +1,159 @@
+# Testing Asynchronous Components
+
+> An example project will be made available.
+
+To simplify testing, `vue-test-utils` applies updates _synchronously_. However, there are some techniques you need to be aware of when testing a component with asynchronous behavior such as callbacks or promises.
+
+One common cases is components that use `watch`, which updates asynchronously. Below in an example of a component that renders some content based on a boolean value, which is updated using a watcher:
+
+``` js
+<template>
+  <div>
+    <input v-model="text" />
+    <button v-if="showButton">Submit</button>
+  </div>
+</template>
+
+<script>
+  export default {
+    data () {
+      return {
+        text: '',
+        showButton: false
+      }
+    },
+
+    watch: {
+      text () {
+        this.showButton = true
+      }
+    }
+  }
+</script>
+```
+
+This component conditionally renders a submit button, based on whether the user entered some text. Let's see how we might test this:
+
+``` js
+import { shallow } from 'vue-test-utils'
+import Foo from './Foo'
+
+describe('Foo', () => {
+  it('renders a button async when the user enters text', () => {
+    expect.assertions(2)
+    const wrapper = shallow(Foo)
+
+    wrapper.find('input').element.value = 'Value'
+    wrapper.find('input').trigger('input')
+
+    expect(wrapper.vm.showButton).toBe(true)
+    expect(wrapper.find('button').exists()).toBe(true)
+  })
+})
+```
+
+The above test fails - however, when you test it by hand in a browser, it seems fine!  The reason the test is failing is because properties inside of `watch` are updated asynchronously. In this test, the assertion occurs before `nextTick()` is called, so the `button` is still not visible.
+
+Most unit test libraries provide a callback to let the runner know when the test is complete. Jest and Karma both use `done()`. We can use this, in combination with `nextTick()`, to test the component.
+
+
+``` js
+import { shallow } from 'vue-test-utils'
+import Foo from './Foo'
+
+describe('Foo', () => {
+  it('renders a button async when the user enters text', (done) => {
+    expect.assertions(2)
+    const wrapper = mount(Foo)
+
+    wrapper.find('input').element.value = 'Value'
+    wrapper.find('input').trigger('input')
+
+    wrapper.vm.$nextTick(() => {
+      expect(wrapper.vm.showButton).toBe(true)
```

Comment:
If this assertion throws, will the error be caught, or will the test time out? In my experience, it's the latter.

Reply:
Mind you, I've been using mocjha. might be different with jest

Reply:
I think it will time out @callumacrae . What is the best thing to do in this situation?

Reply:
not sure. i made an issue about it here: https://github.com/vuejs/vue-test-utils/issues/213